src/AccountManager.mo

```diff
@@ -360,40 +365,41 @@ module {
     /// This method allows a user to deposit tokens by setting up an allowance on their account with the service
     /// principal as the spender and then calling this method to transfer the allowed tokens.
     public func depositFromAllowance(p : Principal, account : ICRC1.Account, amount : Nat) : async* DepositFromAllowanceResponse {
-      if (amount < minimum(#deposit)) return #err(#TooLowQuantity);
+      if (amount <= ledgerFee_) {
+        log(p, #depositViaAllowanceError(#TooLowQuantity));
+        return #err(#TooLowQuantity);
+      };
 
       let transferResult = await* processAllowance(p, account, amount);
 
-      let originalCredit : Nat = amount - fee(#deposit);
+      let originalCredit : Nat = amount - ledgerFee_;
 
       switch (transferResult) {
         case (#Ok txid) {
-          log(p, #consolidated({ deducted = amount; credited = originalCredit }));
-          log(p, #newDeposit(originalCredit));
+          log(p, #allowanceDrawn({ credited = originalCredit }));
           totalConsolidated_ += originalCredit;
           issue(p, originalCredit);
           return #ok(originalCredit, txid);
         };
         case (#Err(#BadFee { expected_fee })) {
           updateFee(expected_fee);
-          let originalCredit_2 : Nat = Int.abs(Int.min(originalCredit, amount - fee(#deposit)));
+          let originalCredit_2 : Nat = Int.abs(Int.min(originalCredit, amount - ledgerFee_));
```

Comment:
Same as above. `originalCredit_2` is probably not needed. We just retry everything exactly as before after calling `updateFee`.